### PR TITLE
[4.0] Move Private Messages to Users

### PR DESCRIPTION
**Pull request by brianteeman**
As the component menu is now used for components that create some form of site content this Pr moves the private messages component from the Component menu to the users menu.

When testing simply applying the patch will only add the messages menu items to the users menu. It will not remove it from the component menu. As the sql change is data and not structure you cannot use the database fix either. It will have to be a direct update in the sql using phpmyadmin or similar

### CODE REVIEW

**Comment by Quy** [comment]
Add module to the Users Dashboard.

**Comment by brianteeman** [comment]
@quy that is beyond the scope of this PR as it doesn't currently exist and there are other things missing from the dashboard

**Comment by brianteeman** [comment]
@SniperSister @zero-24 could you look at rips please

**Comment by zero-24** [comment]
hmm drone seams to fail on system-tests-mysql not rips: https://ci.joomla.org/joomla/joomla-cms/26172 ;)

**Comment by brianteeman** [comment]
This was the failure https://ci.joomla.org/joomla/joomla-cms/26146 @zero-24

**Comment by zero-24** [comment]
Was an false positive it is passing now.

**Comment by brianteeman** [comment]
Conflicts resolved

**Comment by brianteeman** [comment]
Updated branch to kickstart drone

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on ec0c5731000801be2267e467b3c13bed1d279116<br /><br />New messaging menu items appear in the users menu and function as expected. As mentioned, it is also still located in components.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27096">issues.joomla.org/tracker/joomla-cms/27096</a>.</sub>

File: installation/sql/mysql/joomla.sql [hunk 1]
```diff
@@ -1344,27 +1344,21 @@ SELECT 8, 'main', 'com_contact_contacts', 'Contacts', '', 'Contacts/Contacts', '
...
+SELECT 10, 'main', 'com_newsfeeds', 'News Feeds', '', 'News Feeds', 'index.php?option=com_newsfeeds', 'component', 1, 1, 1, `extension_id`, 0, NULL, 0, 0, 'class:newsfeeds', 0, '', 25, 30, 0, '*', 1, NULL, NULL FROM `#__extensions` WHERE `name` = 'com_newsfeeds';
 INSERT INTO `#__menu` (`id`, `menutype`, `title`, `alias`, `note`, `path`, `link`, `type`, `published`, `parent_id`, `level`, `component_id`, `checked_out`, `checked_out_time`, `browserNav`, `access`, `img`, `template_style_id`, `params`, `lft`, `rgt`, `home`, `language`, `client_id`, `publish_up`, `publish_down`)
-SELECT 11, 'main', 'com_messages_add', 'New Private Message', '', 'Messaging/New Private Message', 'index.php?option=com_messages&task=message.add', 'component', 1, 10, 2, `extension_id`, 0, NULL, 0, 0, 'class:messages-add', 0, '', 22, 23, 0, '*', 1, NULL, NULL FROM `#__extensions` WHERE `name` = 'com_messages';
+SELECT 11, 'main', 'com_newsfeeds_feeds', 'Feeds', '', 'News Feeds/Feeds', 'index.php?option=com_newsfeeds&view=newsfeeds', 'component', 1, 12, 2, `extension_id`, 0, NULL, 0, 0, 'class:newsfeeds', 0, '', 26, 27, 0, '*', 1, NULL, NULL FROM `#__extensions` WHERE `name` = 'com_newsfeeds';
```
Comment (Quy) [inline installation/sql/mysql/joomla.sql]: Please check as there are no menu items under News Feeds.

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on cc6474d12aa1646ca539aebdc3f25bc32f4fc4d2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27096">issues.joomla.org/tracker/joomla-cms/27096</a>.</sub>

File: installation/sql/mysql/joomla.sql (hunk 1 above)
Comment (Quy) [inline installation/sql/mysql/joomla.sql]:
```suggestion
SELECT 11, 'main', 'com_newsfeeds_feeds', 'Feeds', '', 'News Feeds/Feeds', 'index.php?option=com_newsfeeds&view=newsfeeds', 'component', 1, 10, 2, `extension_id`, 0, NULL, 0, 0, 'class:newsfeeds', 0, '', 26, 27, 0, '*', 1, NULL, NULL FROM `#__extensions` WHERE `name` = 'com_newsfeeds';
```

File: installation/sql/mysql/joomla.sql [hunk 2]
```diff
@@ -1344,27 +1344,21 @@ SELECT 8, 'main', 'com_contact_contacts', 'Contacts', '', 'Contacts/Contacts', '
...
 INSERT INTO `#__menu` (`id`, `menutype`, `title`, `alias`, `note`, `path`, `link`, `type`, `published`, `parent_id`, `level`, `component_id`, `checked_out`, `checked_out_time`, `browserNav`, `access`, `img`, `template_style_id`, `params`, `lft`, `rgt`, `home`, `language`, `client_id`, `publish_up`, `publish_down`)
-SELECT 12, 'main', 'com_newsfeeds', 'News Feeds', '', 'News Feeds', 'index.php?option=com_newsfeeds', 'component', 1, 1, 1, `extension_id`, 0, NULL, 0, 0, 'class:newsfeeds', 0, '', 25, 30, 0, '*', 1, NULL, NULL FROM `#__extensions` WHERE `name` = 'com_newsfeeds';
+SELECT 12, 'main', 'com_newsfeeds_categories', 'Categories', '', 'News Feeds/Categories', 'index.php?option=com_categories&view=categories&extension=com_newsfeeds', 'component', 1, 12, 2, `extension_id`, 0, NULL, 0, 0, 'class:newsfeeds-cat', 0, '', 24, 25, 0, '*', 1, NULL, NULL FROM `#__extensions` WHERE `name` = 'com_categories';
```
Comment (Quy) [inline installation/sql/mysql/joomla.sql]:
```suggestion
SELECT 12, 'main', 'com_newsfeeds_categories', 'Categories', '', 'News Feeds/Categories', 'index.php?option=com_categories&view=categories&extension=com_newsfeeds', 'component', 1, 10, 2, `extension_id`, 0, NULL, 0, 0, 'class:newsfeeds-cat', 0, '', 24, 25, 0, '*', 1, NULL, NULL FROM `#__extensions` WHERE `name` = 'com_categories';
```

File: installation/sql/postgresql/joomla.sql [hunk 3]
```diff
@@ -1346,27 +1346,21 @@ SELECT 8, 'main', 'com_contact_contacts', 'Contacts', '', 'Contacts/Contacts', '
...
+SELECT 10, 'main', 'com_newsfeeds', 'News Feeds', '', 'News Feeds', 'index.php?option=com_newsfeeds', 'component', 1, 1, 1, "extension_id", 0, NULL, 0, 0, 'class:newsfeeds', 0, '', 25, 30, 0, '*', 1, NULL, NULL FROM "#__extensions" WHERE "name" = 'com_newsfeeds';
 INSERT INTO "#__menu" ("id", "menutype", "title", "alias", "note", "path", "link", "type", "published", "parent_id", "level", "component_id", "checked_out", "checked_out_time", "browserNav", "access", "img", "template_style_id", "params", "lft", "rgt", "home", "language", "client_id", "publish_up", "publish_down")
-SELECT 11, 'main', 'com_messages_add', 'New Private Message', '', 'Messaging/New Private Message', 'index.php?option=com_messages&task=message.add', 'component', 1, 10, 2, "extension_id", 0, NULL, 0, 0, 'class:messages-add', 0, '', 22, 23, 0, '*', 1, NULL, NULL FROM "#__extensions" WHERE "name" = 'com_messages';
+SELECT 11, 'main', 'com_newsfeeds_feeds', 'Feeds', '', 'News Feeds/Feeds', 'index.php?option=com_newsfeeds&view=newsfeeds', 'component', 1, 12, 2, "extension_id", 0, NULL, 0, 0, 'class:newsfeeds', 0, '', 26, 27, 0, '*', 1, NULL, NULL FROM "#__extensions" WHERE "name" = 'com_newsfeeds';
```
Comment (Quy) [inline installation/sql/postgresql/joomla.sql]:
```suggestion
SELECT 11, 'main', 'com_newsfeeds_feeds', 'Feeds', '', 'News Feeds/Feeds', 'index.php?option=com_newsfeeds&view=newsfeeds', 'component', 1, 10, 2, "extension_id", 0, NULL, 0, 0, 'class:newsfeeds', 0, '', 26, 27, 0, '*', 1, NULL, NULL FROM "#__extensions" WHERE "name" = 'com_newsfeeds';
```

File: installation/sql/postgresql/joomla.sql [hunk 4]
```diff
@@ -1346,27 +1346,21 @@ SELECT 8, 'main', 'com_contact_contacts', 'Contacts', '', 'Contacts/Contacts', '
...
 INSERT INTO "#__menu" ("id", "menutype", "title", "alias", "note", "path", "link", "type", "published", "parent_id", "level", "component_id", "checked_out", "checked_out_time", "browserNav", "access", "img", "template_style_id", "params", "lft", "rgt", "home", "language", "client_id", "publish_up", "publish_down")
-SELECT 12, 'main', 'com_newsfeeds', 'News Feeds', '', 'News Feeds', 'index.php?option=com_newsfeeds', 'component', 1, 1, 1, "extension_id", 0, NULL, 0, 0, 'class:newsfeeds', 0, '', 25, 30, 0, '*', 1, NULL, NULL FROM "#__extensions" WHERE "name" = 'com_newsfeeds';
+SELECT 12, 'main', 'com_newsfeeds_categories', 'Categories', '', 'News Feeds/Categories', 'index.php?option=com_categories&view=categories&extension=com_newsfeeds', 'component', 1, 12, 2, "extension_id", 0, NULL, 0, 0, 'class:newsfeeds-cat', 0, '', 28, 29, 0, '*', 1, NULL, NULL FROM "#__extensions" WHERE "name" = 'com_categories';
```
Comment (Quy) [inline installation/sql/postgresql/joomla.sql]:
```suggestion
SELECT 12, 'main', 'com_newsfeeds_categories', 'Categories', '', 'News Feeds/Categories', 'index.php?option=com_categories&view=categories&extension=com_newsfeeds', 'component', 1, 10, 2, "extension_id", 0, NULL, 0, 0, 'class:newsfeeds-cat', 0, '', 28, 29, 0, '*', 1, NULL, NULL FROM "#__extensions" WHERE "name" = 'com_categories';
```

**Comment by brianteeman** [comment]
I have committed your changes @Quy 

I will now leave contributing to joomla 4 until it is clear what is happening with the other template. If that proceeds the way that they want then any change here is a complete waste of all of our time

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 8084ec73f03e936e4d79f2ad8168200ca2ab306d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27096">issues.joomla.org/tracker/joomla-cms/27096</a>.</sub>

**Comment by infograf768** [comment]
Looks like the change is not done for the alternate preset

**Comment by brianteeman** [comment]
Until a decision is made https://github.com/joomla/joomla-cms/pull/27096#issuecomment-570936724 I am not wasting my time

**Comment by marcodings** [comment]
> I will now leave contributing to joomla 4 until it is clear what is happening with the other template. If that proceeds the way that they want then any change here is a complete waste of all of our time

@brianteeman , i to understand what is unclear. The read-me for Khonsu is perfectly clear and has been discussed also with you.

The Khonsu community initiative is intended to be an alternate/alternative template, J4 is NOT waiting for Khonsu, Khonsu will have to live up to the same standards as other contributions. If and only if all these requirements are met it will be considered.

**Comment by brianteeman** [comment]
@marcodings I know we have discussed it before and I know you have stated that before. But I guess you need to tell the people on the Khonsu repository that then. They are definitely still of the opinion that they are building a replacement template. 

See https://github.com/joomla-projects/j4adminui/issues/184#issuecomment-570926024

> Well, Joomla should not have any alternative admin templates. I talked to Marco on this few days back. We have to make our mind fixed. Choose Atum or Khonsu but not both. As an extension developer, I can guarantee you that multiple templates will ruin everything.

>  Keeping backward compatibility with Atum has nothing to do with the requirements. Atum is not a standard.

**Comment by brianteeman** [comment]
> Looks like the change is not done for the alternate preset
@infograf768 

I really don't understand that code. Can you either do a pr to my branch for that or create a separate PR just for that

**Comment by infograf768** [comment]
@brianteeman 
> I really don't understand that code. Can you either do a pr to my branch for that or create a separate PR just for that.

Can do a PR after this one is merged.

**Comment by infograf768** [comment]
Or you can use this .diff in your PR

[messaging.diff.zip](https://github.com/joomla/joomla-cms/files/4030359/messaging.diff.zip)

**Comment by brianteeman** [comment]
Thanks @infograf768 I have applied your patch

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 85e17585bf5748973b24dccb84c405c00aeadd0f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27096">issues.joomla.org/tracker/joomla-cms/27096</a>.</sub>

**Comment by brianteeman** [comment]
I will try to get the conflicts resolved tonight


> @brianteeman , i to understand what is unclear. The read-me for Khonsu is perfectly clear and has been discussed also with you.

Guess that wasn't the case after all. ;)

**Comment by brianteeman** [comment]
@richard67 if you are able to help resolving the conflicts with the sql I would appreciate it. I always mess up on this part.

**Comment by richard67** [comment]
@brianteeman Tonight after work I can have a look on it if nobody else was faster.

**Comment by brianteeman** [comment]
@richard67 greatly appreciated

**Comment by richard67** [comment]
@brianteeman Am on it, stay tuned.

**Comment by richard67** [comment]
@brianteeman Please check [https://github.com/brianteeman/joomla-cms/pull/95](https://github.com/brianteeman/joomla-cms/pull/95), especially the instructions in description, and let me know if you have questions. Btw. is there a way to contact you elsewhere, e.g. on Glip, in case if we have to solve things together and discussion on GH would be uncomfortable?

Update: Note that my PR doesn't correct yet the lft and rgt values in the menu table, see description there.

**Comment by richard67** [comment]
@brianteeman It looks as if lft and rgt were messed up even before. Am just looking and working on fixing them.

Update: With "messed up before" I mean the 4.0-dev branch, not your PR as it was before my changes.

**Comment by richard67** [comment]
@brianteeman I have it, lft and rgt, will make PR for you in a bit, stay tuned. Am doing it in the manual way, like learned on German Joomla Day last year.

**Comment by richard67** [comment]
@brianteeman Delet statements from update SQL script executed in PhpMyAdmin don't delete the old records. Must be something wrong with the where clause. Am checking.

**Comment by richard67** [comment]
@brianteeman It is the 'index.php?' missing at the beginning of the link in the where clause of the update sql script which makes deletion fail.

**Comment by brianteeman** [comment]
oops

File: administrator/components/com_admin/sql/updates/mysql/4.0.0-2019-11-19.sql [hunk 5]
```diff
@@ -0,0 +1,3 @@
+DELETE FROM `#__menu` WHERE `link` = 'index.php?option=com_messages' AND `menutype` = 'main';
+DELETE FROM `#__menu` WHERE `link` = 'index.php?option=com_messages&view=messages' AND `menutype` = 'main';
+DELETE FROM `#__menu` WHERE `link` = 'index.php?option=com_messages&&task=message.add' AND `menutype` = 'main';
```
Comment (richard67) [inline administrator/components/com_admin/sql/updates/mysql/4.0.0-2019-11-19.sql]:
```suggestion
DELETE FROM `#__menu` WHERE `link` = 'index.php?option=com_messages&task=message.add' AND `menutype` = 'main';
```

File: administrator/components/com_admin/sql/updates/postgresql/4.0.0-2019-11-19.sql [hunk 6]
```diff
@@ -0,0 +1,3 @@
+DELETE FROM "#__menu" WHERE "link" = 'index.php?option=com_messages' AND "menutype" = 'main';
+DELETE FROM "#__menu" WHERE "link" = 'index.php?option=com_messages&view=messages' AND "menutype" = 'main';
+DELETE FROM "#__menu" WHERE "link" = 'index.php?option=com_messages&&task=message.add' AND "menutype" = 'main';
```
Comment (richard67) [inline administrator/components/com_admin/sql/updates/postgresql/4.0.0-2019-11-19.sql]:
```suggestion
DELETE FROM "#__menu" WHERE "link" = 'index.php?option=com_messages&task=message.add' AND "menutype" = 'main';
```

**Review by richard67 — CHANGES_REQUESTED** [review]
Final one I think.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on f2a3ccf473a9710289f1a4daaf3098b5a4ab03d0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27096">issues.joomla.org/tracker/joomla-cms/27096</a>.</sub>

**Comment by richard67** [comment]
Tested new install and statements from update sql scripts on MySQL and PostgreSQL. Tested applying the patch on MySQL.

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on f2a3ccf473a9710289f1a4daaf3098b5a4ab03d0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27096">issues.joomla.org/tracker/joomla-cms/27096</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27096">issues.joomla.org/tracker/joomla-cms/27096</a>.</sub>

**Comment by brianteeman** [comment]
thanks all

**Pull request merged**